src/kern/npf_alg_pptp.c

```diff
@@ -125,6 +125,10 @@ typedef struct {
 	/* ... */
 } __packed pptp_outgoing_call_reply_t;
 
+#define PPTP_MIN_MSG_SIZE (MIN(\
```

Comment:
I see the optimisation you are making, but I think such things are potentially error-prone (suppose one day this code will evolve to access a member of the larger struct and that member would b past the size of the smaller struct member). If you want to simpify and move out that `nbuf_ensure_contig()` out of the switch statement, then how about something like this:
```c
#define PPTP_MAX_MSG_SIZE (MAX(\
	sizeof(pptp_outgoing_call_req_t) - sizeof(pptp_msg_hdr_t), \
	sizeof(pptp_outgoing_call_reply_t) - sizeof(pptp_msg_hdr_t)))

        ...

	pptp = nbuf_advance(nbuf, npc->npc_hlen + ((unsigned)th->th_off << 2),
	    sizeof(pptp_msg_hdr_t));
	if (pptp == NULL)
		return false;
        pptp = nbuf_ensure_contig(nbuf, MIN(pptp->len, PPTP_MAX_MSG_SIZE));
        if (pptp == NULL)
                return false;

        ...
```

Otherwise, I think it is not worth to sacrifice the simplicity and correctness of the code for a micro-optimisation here.